pkg/languages/js/js.go

```diff
@@ -130,24 +132,22 @@ func (j jsLanguageManager) DownloadRuntimeIfRequired() error {
 	return nil
 }
 
-func (r jsLanguageManager) ConfigureEditingDirectory(mainFile string, functionConfiguration map[string][]string, editingDirectory string) (string, string, error) {
+// ConfigureEditingDirectory links the function index.js and package.json files to a directory for editing
+func (j jsLanguageManager) ConfigureEditingDirectory(mainFile string, functionConfiguration map[string][]string, editingDirectory string) (string, string, error) {
 	functionFile := path.Join(editingDirectory, "index.js")
 	err := util.Link(mainFile, functionFile)
 	if err != nil {
 		return "", "", err
 	}
 
-	packageJson, err := generatePackageJson(functionConfiguration)
-	if err != nil {
-		return "", "", err
-	}
-
-	err = util.WriteFiles(editingDirectory, util.WriteDest{Filename: "package.json", Data: packageJson})
+	packageJSONFile := path.Join(path.Dir(mainFile), "package.json")
+	packageJSONLink := path.Join(editingDirectory, "package.json")
+	err = util.Link(packageJSONFile, packageJSONLink)
 	if err != nil {
 		return "", "", err
 	}
 
-	return editingDirectory, path.Join(editingDirectory, "package.json"), nil
+	return editingDirectory, functionFile, nil
```

Comment:
Note: I changed the file returned here to be the `index.js` file because that seems like what should be happening here.

Reply:
The second parameter should return the path of the build descriptor (https://github.com/openshift-cloud-functions/kfn/blob/master/pkg/languages/language_manager.go#L19). This is because I initially thought that `idea` expects as cli parameter the build descriptor path, but looks like it accepts the project directory. https://github.com/openshift-cloud-functions/kfn/blob/master/pkg/editors/editor.go#L30

Reply:
Based on the code in `editor.go` it looks like the second return value could be anything. :)

Reply:
yeap exactly for the reason i explained above :) probably we can eliminate it in future